### PR TITLE
Devel mec1501 rtos timer load fix 1

### LINKED ISSUE #17897: k_busy_wait not working when using 32KHz timer driver
k_busy_wait results in inaccurate delays causing MEC1501 I2C driver to always report timeout.
This breaks I2C tests in this board. 

**To Reproduce**
1. mkdir build; cd build
2.  cmake -DBOARD=mec15xxevb_assy6853 
3. make

**Expected behavior**
k_busy_wait should result in expected delays. 

**Impact**
This affects behavior from drivers relaying in k_busy_wait.
Additionally usage of I2C driver cause most tests to fail

**Additional context**

### PR DESCRIPTION
**Pull request by scottwcpg**
Latest MEC1501 RTOS timer changes. Works around HW taking up to one 32KHz cycle to move load register value into count register. Driver has been tested by MCHP and Intel.

Closes #17897

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]

**Review by franciscomunoz — APPROVED** [review]

**Review by franciscomunoz — COMMENTED** [review]
squash your changes in a single commit

File: drivers/timer/mchp_xec_rtos_timer.c [hunk 1]
```diff
@@ -278,6 +291,19 @@ u32_t z_timer_cycle_get_32(void)
 	return ret;
 }
 
+void z_clock_idle_exit(void)
```
Comment (albertofloyd) [inline drivers/timer/mchp_xec_rtos_timer.c]: Should these blocks be under ifdef CONFIG_SYS_POWER_MANAGEMENT ?

**Review by albertofloyd — APPROVED** [review]

**Comment by scottwcpg** [comment]
Tried to squash the two commits into one.

**Review by galak — COMMENTED** [review]
Is this issue board or SoC specific?  (Seems SoC based on the commit message).  As such I'd expect that the config options like `config ARCH_HAS_CUSTOM_BUSY_WAIT` would be in SoC and not board code.

**Pull request merged**